filters/validations.py

```diff
@@ -118,3 +119,69 @@ def fn(value):
                 '<{0}> is not a valid csv of integers'.format(value)
             )
     return fn
+
+
+
+class GenericSeparatedValidator(object):
+    '''
+    Creates list like validator for any voluptuous validation function
+    and any custom string separator.
+    Instance of the class should be passed to the schema like this:
+    >>> CSVofIntegers = GenericSeparatedValidator(int, ',')
+    >>> Schema({ "field": CSVofIntegers() })
+
+    >>> CSVofIntegers = GenericSeparatedValidator(int, ',')
+    >>> CSVofIntegers()('1,2,3')
+    [1,2,3]
+
+    >>> WeirdSeparatedValidation = GenericSeparatedValidator(int, '^^')
+    >>> WeirdSeparatedValidation()('1^^2^^3')
+    [1, 2, 3]
+
+    >>> CSVofIntegerLike = GenericSeparatedValidator(IntegerLike(), ',')
+    >>> CSVofIntegerLike()('a,b,c')
+    Traceback (most recent call last):
+        ...
+    voluptuous.error.Invalid: <a,b,c> is not valid set of <IntegerLike>,\
+    Invalid input <a>; expected an integer
+    '''
+
+    def __init__(self, validation_function, separator=','):
+        if isinstance(separator, basestring):
```

Comment:
This should be `if not isinstance(separator, basestring)`. Maybe add a test for this case?